### ISSUE TITLE
Deploy Realm Object Server to Heroku

### ISSUE THREAD

**Issue by wigging** [realm/realm-mobile-platform#5]
As a feature request, I would like to suggest an option to easily deploy the Realm Object Server to Heroku. I am aware of the AMI image for Amazon services but a Heroku option would be another easy option for deployment. Are there any plans to support Heroku deployment?

**Comment by radu-tutueanu** [comment]
Hello, @wigging , and thanks for submitting this feature request. We're happy to hear that you will be able to get up and running using the Amazon AMI (especially considering that you won't incur too high latency and no bandwidth charges, as long as you deploy in the same AZ). We're still considering our options with regards to Heroku integration.

What exactly would you have in mind? Would you like to see a one-click-button solution as there currently exists for 3rd party services, such as hosted Redis and friends, or would you like to have more manual control, and just want help deploying the ROS to a Heroku worker instance?

 We'd love to hear what kind of use-cases you have in mind, and how exactly you would deploy the ROS in order to help us focus our development efforts.

**Comment by ChrisHampel** [comment]
I would very much like to see this too. We are looking at using Realm Object Server and this is one of the concerns we have, as the rest of our infrastructure is on Heroku already. As for which is better: Self-hosting would be nice simply because then we have control over the server version, however, if you provided the ability to pick the version being run,as part of your hosted service,  say current and 1+ versions back. By being similar to Heroku's Postgres in this way, it would allow Companies time to prepare for version changes.

**Comment by radu-tutueanu** [comment]
Thank you for the input, @ChrisHampel . I can see how this kind of control would be important to you and we will keep that in mind.

**Comment by wigging** [comment]
@radu-tutueanu When developing iOS and Mac apps, I prefer to spend most of my time developing the app and not the server side logic. So I tend to rely on services like Heroku that will maintain the server. The one-click provision would be the ideal situation.

**Comment by jaredmichaelsmith** [comment]
I would love to see this happen!

**Comment by kerbelda** [comment]
We, at Salesforce (and by extension Heroku), would love to see this :), or at least an API for the Realm Server Object. At the moment we can't sync data from the Realm Server Object in Amazon to anything outside realm like Heroku's Postgres. Please let me know if we can connect you with anyone.

**Comment by bigfish24** [comment]
@kerbelda Adam, Director of Product, I would love to chat. Could you email me at af@realm.io? I have background with Salesforce/Heroku/Heroku Connect so definitely would love to see how we could help.

**Comment by john-brunelle** [comment]
+1 This something that I would love to see happen in the near future.

**Comment by danieldelouya** [comment]
+1

**Comment by ynagarjuna1995** [comment]
+1

**Comment by jschoolcraft** [comment]
Any updates on when/if this is happening?

**Comment by scottfister** [comment]
Would love to see one-click install for Heroku!

**Comment by fpatelm** [comment]
I have installed the RealmObject server using the commands for Ubuntu on AWS, is there a easy way? If anyone wants to share some guides on it. Anyway With Ubuntu configs I manage to have it run.

**Comment by sirvon** [comment]
@fpatelm can you share how you got it done?

**Comment by arminasr** [comment]
+1
Installation of Realm Object Server on Heroku is the feature that I want!

**Comment by NSMyself** [comment]
+1

**Comment by ArifADS** [comment]
How can it be installed on Heroku if Heroku doesn't persist data in dynos restart?

**Comment by jpsim** [comment]
It would presumably have to be provided as a Heroku Add-On, much like other database providers like [mLab](https://elements.heroku.com/addons/mongolab).

**Comment by ArifADS** [comment]
@jpsim forgive me if i'm wrong, but mLab offers they own servers to store the MongoDB data. The Add On basically only creates the accounts, mongo instance, set env variables and links it to your Heroku Account.

While Realm doesn't offer they servers to store the Realm Data. But, I actually don't know if a Heroku Add-On has the ability to store persistent data in file system.

**Comment by jpsim** [comment]
Realm would have to run servers to provide a Heroku add-on.

**Comment by viksgyl** [comment]
You can try my docker image. I am able to run it on openshift online.
https://hub.docker.com/r/viksgyl/realm-object-server/

**Comment by bigfish24** [comment]
No, it is simply not possible to run ROS on Heroku. We would need to provide a cloud service that is simply sold through the Heroku marketplace as an add-on as mentioned above. This could happen but I don't have any specific dates I can share.

**Comment by regnatarajan** [comment]
I would be interested in ROS as a service regardless of whether it is on Heroku or some other system.  I simply don't want to maintain servers at this point and I'm willing to pay a service to do it for me. They can sweat the uptime challenges instead of me.  I love the AWS model of only charging for usage with Lambda/DynamoDB and ideally I'd love to see an offering like that from Realm.

**Comment by amirdew** [comment]
I just set the port for Heroku (```process.env.PORT```) and it's worked, ```src/index.ts```:
```import { BasicServer } from 'realm-object-server'
import * as path from 'path'

const server = new BasicServer()

server.start({ 
        dataPath: path.join(__dirname, '../data'), 
        port: process.env.PORT || 9080 
    })
    .then(() => {
        console.log(`Realm Object Server was started on ${server.address}`)
    })
    .catch(err => {
        console.error(`Error starting Realm Object Server: ${err.message}`)
    })
```
